lcls_tools/common/measurements/screen_beam_profile_measurement.py

```diff
@@ -0,0 +1,99 @@
+from lcls_tools.common.devices.screen import Screen
+from lcls_tools.common.image.processing import ImageProcessor
+from lcls_tools.common.data.fit.projection import ProjectionFit
+from lcls_tools.common.measurements.measurement import Measurement
+import numpy as np
+
+
+class ScreenBeamProfileMeasurement(Measurement):
+    """
+    ScreenBeamProfileMeasurement class that allows for background subtraction and roi cropping
+    ------------------------
+    Arguments:
+    name: str (name of measurement default is beam_profile),
+    device: Screen (device that will be performing the measurement),
+    image_processor: ImageProcessor ()
+    fitting_tool: ProjectionFit ()
+    fit_profile: bool = True ()
+    ------------------------
+    Methods:
+    single_measure: measures device and returns raw and processed image
+    measure: does multiple measurements and has an option to fit the image profiles
+    """
+
+    name: str = "beam_profile"
+    device: Screen
+    image_processor: ImageProcessor
+    fitting_tool: ProjectionFit
+    fit_profile: bool = True
+    # charge_window: Optional[ChargeWindow] = None
+
+    def measure(self, n_shots: int = 1) -> dict:
+        """
+        Measurement function that takes in n_shots as argumentment, where n_shots is the number of
+        image profiles (is this what I want to call it?)
+        we would like to measure. Invokes single_measure per shot
+        Then if fit_profile = True, fits the x and y projections of each image using the provided
+        fitting_tool. Results are stored in list of dictionaries where each element of the list
+        is results for the image fitting process stored as a dictionary.
+        ----
+        Currently under work, what do we want to do with the images?
+        Needs dump_controller
+
+        """
+        images = []
+        while len(images) < n_shots:
+            measurement = self.single_measure()
+            if len(measurement):
```

Comment:
If a measurement returns a bad image while len(image) < n_shots, there's no catch here?  i.e. if the 3rd out of 10 image measurement returns nothing, it keeps going but would only return 9 good images? Or maybe I'm reading that wrong?

Reply:
As the code is written right now, it will keep trying to grab an image until it has n images, regardless of the result. I assume this would only happen if something went wrong in CA. How are we handling CA errors in devices right now?

Reply:
Ah yes, I see what you mean. My holiday brain...
Were not doing any sophisticated error handling. A few places with PV checks and attribute errors. 
So it would cause problems if there was an epics error returned/we need to look into that in future PRs.